packages/camera/camera_web/lib/src/camera_service.dart

```diff
@@ -393,4 +394,68 @@ class CameraService {
         return DeviceOrientation.portraitUp;
     }
   }
+
+  /// Maps a [Uint] to a [CameraImageData].
+  CameraImageData getCameraImageDataFromBytes(
+    Uint8List bytes, {
+    required int height,
+    required int width,
+  }) {
+    return CameraImageData(
+      format: const CameraImageFormat(
+        ImageFormatGroup.jpeg,
+        raw: '',
+      ),
+      planes: <CameraImagePlane>[
+        CameraImagePlane(
+          bytes: bytes,
+          bytesPerRow: 0,
+        ),
+      ],
+      height: height,
+      width: width,
+    );
+  }
+
+  ///Returns frame at a specific time using video element
+  CameraImageData takeFrame(html.VideoElement videoElement) {
+    final List<String> widthPx = videoElement.style.width.split('px');
+    final List<String> heightPx = videoElement.style.height.split('px');
+    final String widthString =
+        widthPx.isNotEmpty ? widthPx.first : '${videoElement.videoWidth}';
+    final String heightString =
+        heightPx.isNotEmpty ? heightPx.first : '${videoElement.videoHeight}';
+    final int width = int.tryParse(widthString) ?? videoElement.videoWidth;
+    final int height = int.tryParse(heightString) ?? videoElement.videoHeight;
```

Comment:
Please, try using the `int videoElement.videoWidth/videoHeight` before doing all the dance of parsing strings.

Also: does this need to be determined on every captured frame? Can we do it in a resize listener of the video element and only recompute this when it changes?

Also also: the `videoWidth/videoHeight` doesn't appear to be nullable; do we really need to do any string parsing? is the `style` version more accurate than the reported `videoWidth/Height`?

Reply:
I checked, the string interpolation is not necessary. Removing it in next commits.
`int videoElement.videoWidth/videoHeight` is a better approach.

Reply:
Apologies for that code,
We do not need to determine `videoWidth/videoHeight` on every frame, because these parameter never changes during the life-cyle of the `videoElement`. It only changes when camera is initialized.

Here's what I found in `camera_web.dart` file
```
  /// Emits an empty stream as there is no event corresponding to a change
  /// in the camera resolution on the web.
  ///
  /// In order to change the camera resolution a new camera with appropriate
  /// [CameraOptions.video] constraints has to be created and initialized.
  @override
  Stream<CameraResolutionChangedEvent> onCameraResolutionChanged(int cameraId) {
    return const Stream<CameraResolutionChangedEvent>.empty();
  }
```

Reply:
And something like this can be added to listen to resize changes on videoElement.

```
 final Map<int, StreamSubscription<html.Event>>
      _cameraVideoResizedSubscriptions =
      <int, StreamSubscription<html.Event>>{};
```
```
      // Add camera's resolution events to the camera events stream.
      // The onResize stream fires when videoElement gets resized
      _cameraVideoResizedSubscriptions[cameraId] =
          camera.videoElement.onResize.listen((html.Event _) {
        cameraEventStreamController.add(
          CameraResolutionChangedEvent(
            cameraId,
            camera.videoElement.width.toDouble(),
            camera.videoElement.height.toDouble(),
          ),
        );
      });
```
```
  @override
  Stream<CameraResolutionChangedEvent> onCameraResolutionChanged(int cameraId) {
    return _cameraEvents(cameraId).whereType<CameraResolutionChangedEvent>();
  }
```

But it will be non functional due to the reason mentioned in the above comment's code!